### ISSUE TITLE
Dualsense Controller not Working With Duo

### ISSUE THREAD

**Issue by AnshulJ999** [DuoStream/Duo#204]
Hi, might be a noob question, but I've waded through all the issues and Reddit discussions around Duo and could not find a solution to this: 

I have a Dualsense connected to my main laptop (host, Win 10). I tried both Bluetooth and USB connections. 

When I run Duo, it doesn't recognize this controller. I tried various games and Steam, no dice. I tried connecting the controller first, and then Duo, and also vice versa. 

Basically my Dualsense doesn't work with Duo no matter what I've tried. 

I read a lot about 'controller jailing' here, but I hardly understand it. 

My client is: Android 13 tablet with the Moonlight client. 

Another thing is: I selected play audio on both client and host, but the audio doesn't work on the host. 

I have found a workaround solution for now, using scrcpy to pass audio and gamepad to-and-fro the host and client; kinda of an unstable solution so would like to know if I can make it work without that. Appreciate any help, thanks.

**Comment by samkitty** [comment]
Dualsense should work fine through Duo. Have you tried the official Sunshine? Do you have the same issues there?

Try going into Sunshine settings through the web portal button in Duo Configurator and under **input** tab make sure controller input is enabled and set the controller type to Xbox. Then go to the **audio** tab and fill in the audio sink fields for the audio on host to work, otherwise the default audio device would be Steamlink Audio which mutes the host speaker.

**Comment by AnshulJ999** [comment]
So I havent tried official Sunshine, but I tried Apollo, which is a direct fork. And yes Dualsense works fine in that case, but that's probably because it's showing the same screen/system as the main laptop. 

I'm not sure what Duo does exactly, but the virtual system created by Duo is not recognizing the Dualsense controller; it's still tied to the primary system. Hope that explains things a bit. 

![image](https://github.com/user-attachments/assets/0d515939-1778-4410-9359-9e48586f3779)

![image](https://github.com/user-attachments/assets/4c7a0f11-f7b2-4c55-9889-5df295035eac)

Here are my settings in the Input tab and in the Audio/Video tab. I specified my main output device, but still no audio :/ 

Am I supposed to change the 'Remote Audio' to something else? 

Also, I specified my NVidia GPU but based on some tests I think Duo is using my Intel GPU; as it goes up to 100% during games while the Nvidia one stays idle. 

So TLDR:

1) Dualsense not recognized
2) Audio not playing through host
3) Seems like dGPU is not being used. Basic games work, but modern games aren't working for me. 

Duo really is a fantastic idea; I'd love to get it working properly and then even support via Patreon honestly.

**Comment by AnshulJ999** [comment]
Update: So for my problems 1 and 2, I was able to find a good combination of Scrcpy arguments that work well for me: `scrcpy -KM --no-video --audio-codec=aac -G`

So this passes the Dualsense from my host laptop to the tablet, and the audio is passed back to the host. I've given up on making it work natively for now; too much time wasted there. 

For 3, I just deleted those arguments from above pics to reset them to default and checked my Nvidia control panel settings; now it seems to be in order and the GPU is used. 

Hope the next updates make all this even easier to do and more robust :D

**Comment by samkitty** [comment]
Your setup is confusing to me. You want to play on a tablet but the audio to come from your laptop? I don't think that is possible because the Duo session runs separately from the console session (host) and it does not interface with the host video/audio hardware. The audio will only come out of the speakers that are connected to your tablet, regardless of the settings you put in Sunshine (so the "play audio on host pc" option in Moonlight is not applicable to Duo).

The Dualsense should work out of the box assuming you are connecting it to your tablet. If it is still controlling the console session (host) then controller jailing isn't working for you, but it should still control the Duo session alongside it regardless. Have you tried connecting it to your tablet both wired/wireless? How about switching the controller type to Xbox rather than DS4 in sunshine settings? Do the default Moonlight on screen touch-controls work? Do you have hidhide installed? Try resetting hidhide config and uninstalling it. Also if you have a different fork of sunshine running in the background you should close it.

I imagine your Scrcpy setup is adding a ton of latency and it should not be needed. @Black-Seraph will be able to provide better help by looking through your logs so I recommend you post them here in advance for when he has enough free time to help.

**Comment by AnshulJ999** [comment]
Thanks for the detailed reply. I realize my use case is a bit out of the ordinary and I misunderstood some stuff, and also explained some things unclearly. 

So my host is my main gaming laptop, but I wanted to play games on my tablet, in the same room, in a different corner. Now my Dualsense is connected to the Host, and not to the client (tablet). I don't really want to keep pairing/unpairing each time I switch from one machine to the other. 

I thought Duo had some in-built way to pass the controller from the Host to the Client, but I was mistaken. It can pass the controller only in one way it seems. 

And as you clarified, the audio is also a no-go. 

So for now, Scrcpy is my only solution really. It's not perfect, but any latency is unnoticeable to me. 

Also I was using the same windows account earlier, but I realize it causes some problems, so I'll need to look into creating a separate one just for gaming on Duo. 

Just for reference here are the logs from my last session:

```
[2024-12-23 03:48:06.990]: Info: Sunshine version: 0.0.0.8b866d5b
[2024-12-23 03:48:06.990]: Info: Package Publisher: Black-Seraph
[2024-12-23 03:48:06.990]: Info: Publisher Website: https://www.black-seraph.com
[2024-12-23 03:48:06.990]: Info: Get support: https://github.com/DuoStream/Duo/issues
[2024-12-23 03:48:07.205]: Info: nvprefs: No need to modify application profile settings
[2024-12-23 03:48:07.205]: Info: nvprefs: Changed OGL_CPL_PREFER_DXPRESENT to OGL_CPL_PREFER_DXPRESENT_PREFER_ENABLED for base profile
[2024-12-23 03:48:07.277]: Info: Compiling shaders...
[2024-12-23 03:48:07.296]: Warning: include/base_vs.hlsl(28,13-36): warning X3556: integer modulus may be much slower, try using uints if possible.
[2024-12-23 03:48:07.322]: Warning: include/base_vs.hlsl(28,13-36): warning X3556: integer modulus may be much slower, try using uints if possible.
[2024-12-23 03:48:07.336]: Warning: include/base_vs.hlsl(28,13-36): warning X3556: integer modulus may be much slower, try using uints if possible.
[2024-12-23 03:48:07.346]: Warning: include/base_vs.hlsl(28,13-36): warning X3556: integer modulus may be much slower, try using uints if possible.
[2024-12-23 03:48:07.373]: Warning: include/base_vs.hlsl(28,13-36): warning X3556: integer modulus may be much slower, try using uints if possible.
[2024-12-23 03:48:07.381]: Warning: include/base_vs.hlsl(28,13-36): warning X3556: integer modulus may be much slower, try using uints if possible.
[2024-12-23 03:48:07.381]: Info: Compiled shaders
[2024-12-23 03:48:07.431]: Info: ViGEmBus device instance path: ROOT\SYSTEM\0001
[2024-12-23 03:48:07.431]: Info: // Testing for available encoders, this may generate errors. You can safely ignore those errors. //
[2024-12-23 03:48:07.431]: Info: Trying encoder [nvenc]
[2024-12-23 03:48:07.488]: Info: ddprobe.exe 1  --verify-frame-capture returned 0x887A0026
[2024-12-23 03:48:07.654]: Info: ddprobe.exe 2  --verify-frame-capture returned 0x00000000
[2024-12-23 03:48:07.654]: Info: Set GPU preference: 2
[2024-12-23 03:48:07.800]: Info: 
Device Description : NVIDIA GeForce RTX 3050 Ti Laptop GPU
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x000025A0
Device Video Mem   : 3962 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 16273 MiB
Feature Level      : 0x0000B100
Capture size       : 1920x1080
Offset             : 0x0
Virtual Desktop    : 1920x1080
[2024-12-23 03:48:07.801]: Info: Active GPU has HAGS enabled
[2024-12-23 03:48:07.801]: Info: Using realtime GPU priority
[2024-12-23 03:48:07.802]: Info: Display refresh rate [30Hz]
[2024-12-23 03:48:07.802]: Info: Requested frame rate [60fps]
[2024-12-23 03:48:07.802]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0,0]
Green Primary      : [0,0]
Blue Primary       : [0,0]
White Point        : [0,0]
Min Luminance      : 0 nits
Max Luminance      : 0 nits
Max Full Luminance : 0 nits
[2024-12-23 03:48:07.802]: Info: Requesting mode change to 1920x1080@60 using the SDR colorspace
[2024-12-23 03:48:07.803]: Info: Creating encoder [h264_nvenc]
[2024-12-23 03:48:07.803]: Info: Color coding: SDR (Rec. 601)
[2024-12-23 03:48:07.803]: Info: Color depth: 8-bit
[2024-12-23 03:48:07.803]: Info: Color range: JPEG
[2024-12-23 03:48:07.847]: Info: NvEnc: created encoder H.264 P1 two-pass rfi
[2024-12-23 03:48:07.872]: Info: Creating encoder [hevc_nvenc]
[2024-12-23 03:48:07.872]: Info: Color coding: SDR (Rec. 601)
[2024-12-23 03:48:07.872]: Info: Color depth: 8-bit
[2024-12-23 03:48:07.872]: Info: Color range: JPEG
[2024-12-23 03:48:07.919]: Info: NvEnc: created encoder HEVC P1 two-pass rfi
[2024-12-23 03:48:07.952]: Info: Creating encoder [av1_nvenc]
[2024-12-23 03:48:07.952]: Info: Color coding: SDR (Rec. 601)
[2024-12-23 03:48:07.952]: Info: Color depth: 8-bit
[2024-12-23 03:48:07.952]: Info: Color range: JPEG
[2024-12-23 03:48:07.975]: Error: NvEnc: encoding format is not supported by the gpu
[2024-12-23 03:48:07.983]: Info: Creating encoder [h264_nvenc]
[2024-12-23 03:48:07.983]: Info: Color coding: SDR (Rec. 601)
[2024-12-23 03:48:07.983]: Info: Color depth: 8-bit
[2024-12-23 03:48:07.983]: Info: Color range: JPEG
[2024-12-23 03:48:08.030]: Info: NvEnc: created encoder H.264 P1 yuv444 two-pass rfi
[2024-12-23 03:48:08.147]: Info: ddprobe.exe 1  --verify-frame-capture returned 0x887A0026
[2024-12-23 03:48:08.320]: Info: ddprobe.exe 2  --verify-frame-capture returned 0x00000000
[2024-12-23 03:48:08.320]: Info: Set GPU preference: 2
[2024-12-23 03:48:08.359]: Info: 
Device Description : NVIDIA GeForce RTX 3050 Ti Laptop GPU
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x000025A0
Device Video Mem   : 3962 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 16273 MiB
Feature Level      : 0x0000B100
Capture size       : 1920x1080
Offset             : 0x0
Virtual Desktop    : 1920x1080
[2024-12-23 03:48:08.359]: Info: Active GPU has HAGS enabled
[2024-12-23 03:48:08.359]: Info: Using realtime GPU priority
[2024-12-23 03:48:08.359]: Info: Display refresh rate [30Hz]
[2024-12-23 03:48:08.359]: Info: Requested frame rate [60fps]
[2024-12-23 03:48:08.359]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0,0]
Green Primary      : [0,0]
Blue Primary       : [0,0]
White Point        : [0,0]
Min Luminance      : 0 nits
Max Luminance      : 0 nits
Max Full Luminance : 0 nits
[2024-12-23 03:48:08.360]: Info: Requesting mode change to 1920x1080@60 using the HDR colorspace
[2024-12-23 03:48:08.360]: Info: Creating encoder [hevc_nvenc]
[2024-12-23 03:48:08.360]: Info: Color coding: SDR (Rec. 709)
[2024-12-23 03:48:08.360]: Info: Color depth: 10-bit
[2024-12-23 03:48:08.360]: Info: Color range: JPEG
[2024-12-23 03:48:08.505]: Info: NvEnc: created encoder HEVC P1 yuv444 10-bit two-pass rfi
[2024-12-23 03:48:08.553]: Info: 
[2024-12-23 03:48:08.553]: Info: // Ignore any errors mentioned above, they are not relevant. //
[2024-12-23 03:48:08.553]: Info: 
[2024-12-23 03:48:08.554]: Info: Found H.264 encoder: h264_nvenc [nvenc]
[2024-12-23 03:48:08.554]: Info: Found HEVC encoder: hevc_nvenc [nvenc]
[2024-12-23 03:48:08.557]: Info: Configuration UI available at [https://localhost:31234]
[2024-12-23 03:48:09.336]: Info: Registered Sunshine mDNS service
[2024-12-23 03:49:07.169]: Info: Encoder reenumeration is required
[2024-12-23 03:49:07.169]: Info: // Testing for available encoders, this may generate errors. You can safely ignore those errors. //
[2024-12-23 03:49:07.170]: Info: Trying encoder [nvenc]
[2024-12-23 03:49:07.244]: Info: ddprobe.exe 1  --verify-frame-capture returned 0x887A0026
[2024-12-23 03:49:07.419]: Info: ddprobe.exe 2  --verify-frame-capture returned 0x00000000
[2024-12-23 03:49:07.419]: Info: Set GPU preference: 2
[2024-12-23 03:49:07.460]: Info: 
Device Description : NVIDIA GeForce RTX 3050 Ti Laptop GPU
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x000025A0
Device Video Mem   : 3962 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 16273 MiB
Feature Level      : 0x0000B100
Capture size       : 1920x1080
Offset             : 0x0
Virtual Desktop    : 1920x1080
[2024-12-23 03:49:07.460]: Info: Active GPU has HAGS enabled
[2024-12-23 03:49:07.461]: Info: Using realtime GPU priority
[2024-12-23 03:49:07.461]: Info: Display refresh rate [30Hz]
[2024-12-23 03:49:07.461]: Info: Requested frame rate [60fps]
[2024-12-23 03:49:07.461]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0,0]
Green Primary      : [0,0]
Blue Primary       : [0,0]
White Point        : [0,0]
Min Luminance      : 0 nits
Max Luminance      : 0 nits
Max Full Luminance : 0 nits
[2024-12-23 03:49:07.461]: Info: Requesting mode change to 1920x1080@60 using the SDR colorspace
[2024-12-23 03:49:07.461]: Info: Creating encoder [h264_nvenc]
[2024-12-23 03:49:07.461]: Info: Color coding: SDR (Rec. 601)
[2024-12-23 03:49:07.461]: Info: Color depth: 8-bit
[2024-12-23 03:49:07.462]: Info: Color range: JPEG
[2024-12-23 03:49:07.506]: Info: NvEnc: created encoder H.264 P1 two-pass rfi
[2024-12-23 03:49:07.531]: Info: Creating encoder [hevc_nvenc]
[2024-12-23 03:49:07.532]: Info: Color coding: SDR (Rec. 601)
[2024-12-23 03:49:07.532]: Info: Color depth: 8-bit
[2024-12-23 03:49:07.532]: Info: Color range: JPEG
[2024-12-23 03:49:07.581]: Info: NvEnc: created encoder HEVC P1 two-pass rfi
[2024-12-23 03:49:07.623]: Info: Creating encoder [av1_nvenc]
[2024-12-23 03:49:07.623]: Info: Color coding: SDR (Rec. 601)
[2024-12-23 03:49:07.623]: Info: Color depth: 8-bit
[2024-12-23 03:49:07.623]: Info: Color range: JPEG
[2024-12-23 03:49:07.648]: Error: NvEnc: encoding format is not supported by the gpu
[2024-12-23 03:49:07.655]: Info: Creating encoder [h264_nvenc]
[2024-12-23 03:49:07.655]: Info: Color coding: SDR (Rec. 601)
[2024-12-23 03:49:07.655]: Info: Color depth: 8-bit
[2024-12-23 03:49:07.655]: Info: Color range: JPEG
[2024-12-23 03:49:07.700]: Info: NvEnc: created encoder H.264 P1 yuv444 two-pass rfi
[2024-12-23 03:49:07.811]: Info: ddprobe.exe 1  --verify-frame-capture returned 0x887A0026
[2024-12-23 03:49:08.012]: Info: ddprobe.exe 2  --verify-frame-capture returned 0x00000000
[2024-12-23 03:49:08.012]: Info: Set GPU preference: 2
[2024-12-23 03:49:08.058]: Info: 
Device Description : NVIDIA GeForce RTX 3050 Ti Laptop GPU
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x000025A0
Device Video Mem   : 3962 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 16273 MiB
Feature Level      : 0x0000B100
Capture size       : 1920x1080
Offset             : 0x0
Virtual Desktop    : 1920x1080
[2024-12-23 03:49:08.058]: Info: Active GPU has HAGS enabled
[2024-12-23 03:49:08.058]: Info: Using realtime GPU priority
[2024-12-23 03:49:08.058]: Info: Display refresh rate [30Hz]
[2024-12-23 03:49:08.058]: Info: Requested frame rate [60fps]
[2024-12-23 03:49:08.059]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0,0]
Green Primary      : [0,0]
Blue Primary       : [0,0]
White Point        : [0,0]
Min Luminance      : 0 nits
Max Luminance      : 0 nits
Max Full Luminance : 0 nits
[2024-12-23 03:49:08.059]: Info: Requesting mode change to 1920x1080@60 using the HDR colorspace
[2024-12-23 03:49:08.059]: Info: Creating encoder [hevc_nvenc]
[2024-12-23 03:49:08.059]: Info: Color coding: SDR (Rec. 709)
[2024-12-23 03:49:08.059]: Info: Color depth: 10-bit
[2024-12-23 03:49:08.059]: Info: Color range: JPEG
[2024-12-23 03:49:08.183]: Info: NvEnc: created encoder HEVC P1 yuv444 10-bit two-pass rfi
[2024-12-23 03:49:08.241]: Info: 
[2024-12-23 03:49:08.241]: Info: // Ignore any errors mentioned above, they are not relevant. //
[2024-12-23 03:49:08.242]: Info: 
[2024-12-23 03:49:08.242]: Info: Found H.264 encoder: h264_nvenc [nvenc]
[2024-12-23 03:49:08.242]: Info: Found HEVC encoder: hevc_nvenc [nvenc]
[2024-12-23 03:49:08.242]: Info: Executing [Desktop]
[2024-12-23 03:49:08.466]: Info: nvprefs: No need to modify application profile settings
[2024-12-23 03:49:08.466]: Info: nvprefs: No need to modify global profile settings
[2024-12-23 03:49:08.481]: Info: WLAN interface 0 is now in low latency mode
[2024-12-23 03:49:08.487]: Info: CLIENT CONNECTED
[2024-12-23 03:49:08.577]: Info: ddprobe.exe 1  --verify-frame-capture returned 0x887A0026
[2024-12-23 03:49:08.584]: Info: Gamepad 0 will be Xbox 360 controller (auto-selected by client-reported type)
[2024-12-23 03:49:08.585]: Info: Jailed USB\VID_045E&PID_028E\01 into session 5
[2024-12-23 03:49:08.585]: Info: Jailed USB\VID_045E&PID_028E\01 into session 5
[2024-12-23 03:49:08.593]: Info: Jailed USB\VID_045E&PID_028E&IG_00\2&DEE0F28&7&00 into session 5
[2024-12-23 03:49:08.800]: Info: Audio capture format is [F32 48000 2.0]
[2024-12-23 03:49:08.802]: Info: Opus initialized: 48 kHz, 2 channels, 96 kbps (total), LOWDELAY
[2024-12-23 03:49:08.907]: Info: ddprobe.exe 2  --verify-frame-capture returned 0x00000000
[2024-12-23 03:49:08.907]: Info: Set GPU preference: 2
[2024-12-23 03:49:09.107]: Info: ddprobe.exe 1 \\.\DISPLAY1 --verify-frame-capture returned 0x887A0026
[2024-12-23 03:49:09.392]: Info: ddprobe.exe 2 \\.\DISPLAY1 --verify-frame-capture returned 0x00000000
[2024-12-23 03:49:09.393]: Info: Set GPU preference: 2
[2024-12-23 03:49:09.433]: Info: 
Device Description : NVIDIA GeForce RTX 3050 Ti Laptop GPU
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x000025A0
Device Video Mem   : 3962 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 16273 MiB
Feature Level      : 0x0000B100
Capture size       : 1920x1080
Offset             : 0x0
Virtual Desktop    : 1920x1080
[2024-12-23 03:49:09.433]: Info: Active GPU has HAGS enabled
[2024-12-23 03:49:09.433]: Info: Using realtime GPU priority
[2024-12-23 03:49:09.433]: Info: Display refresh rate [30Hz]
[2024-12-23 03:49:09.433]: Info: Requested frame rate [30fps]
[2024-12-23 03:49:09.433]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0,0]
Green Primary      : [0,0]
Blue Primary       : [0,0]
White Point        : [0,0]
Min Luminance      : 0 nits
Max Luminance      : 0 nits
Max Full Luminance : 0 nits
[2024-12-23 03:49:09.434]: Info: Requesting mode change to 1920x1080@30 using the SDR colorspace
[2024-12-23 03:49:09.434]: Info: Creating encoder [h264_nvenc]
[2024-12-23 03:49:09.434]: Info: Color coding: SDR (Rec. 709)
[2024-12-23 03:49:09.434]: Info: Color depth: 8-bit
[2024-12-23 03:49:09.434]: Info: Color range: MPEG
[2024-12-23 03:49:09.483]: Info: NvEnc: created encoder H.264 P1 two-pass rfi
[2024-12-23 04:22:27.437]: Info: Creating virtual touch input device
[2024-12-23 04:22:29.405]: Info: CLIENT DISCONNECTED
```

I didn't see any problems as such. Thanks for the help :D

**Comment by samkitty** [comment]
It sounds like Duo isn't exactly what you need in that case. Do you need the laptop to be displaying something else while you game? Is that why you replaced Apollo with Duo? If that is the case you should look into using a virtual multi-monitor sunshine setup, where the main display is a virtual adapter that is streamed to your tablet and the physical laptop screen is just a second monitor that extends your desktop so you can have things displayed on it in the background while you game on the tablet. If that isn't your use case then I honestly still don't understand why you're not just using normal sunshine since everything is going through the laptop anyway haha

**Comment by AnshulJ999** [comment]
Actually Duo is perfect for what I'm about to do and I didn't even think it was possible but I'm amazed that it actually works. The tablet thing is just a test run for now; what I wanted is for my laptop to be fully usable while gaming. 

So my brother can play games while I'm working, and I can play games while someone else is using the laptop. Heck I can only imagine what else this might be useful for; a virtual instance of my laptop that can function independently (mostly) could be used for a lot more than games I think. As I see it that's Duo's core functionality and perfectly suits me; Sunshine/Apollo/others don't have that. 

I've been testing Apollo just to see if it works better when streaming to my TV; when I don't need my laptop free. But my network setup kinda sucks so the TV thing hasn't worked well yet. 

But thank you for the help, and big thanks to Black-Seraph for this amazing software that I didn't even think could exist.

**Comment by samkitty** [comment]
Mhm, Duo is pretty much magic. Definitely consider subscribing on Patreon to support the (solo) dev and get extra features like higher refresh rate.

I'm glad you got things figured out so this issue can be closed for now.

Good luck with your future setup and Merry Christmas! ᵔ⤙ᵔ

**Comment by AnshulJ999** [comment]
Hi, just wanted to ask a few more queries here instead of opening a new issue: 

1) Is there any way to pass the Dualsense as it is; for Playstation games like GoW Ragnarok and Horizon, which support it natively? I mean right now I see only a 'DS4' and 'Xbox' option. I'm not sure if original Sunshine supports this yet either, but it'd be a nice feature to have. 

2) Is there any chance we'll see Win11 exclusive features like Sandboxing available on Win10 as well? I mean is it a technical limitation preventing that? As of now the only premium features I saw on Win 10 are higher refresh rate and more instances; the rest is only for Win11. 

And merry Christmas!

**Comment by samkitty** [comment]
1- This is a Sunshine limitation outside of Duo's control. DS4 and Xbox controller drivers have been extensively reverse engineered and open source emulators are very common. But that's is not the case for Dualsense yet. The only software that is capable of emulating a Dualsense controller is DualsenseX, which you can find on Steam (paid). Not sure whether passing DS4 input then forcing it to be recognized as Dualsense through DSX would be possible but it's worth trying. You might need HidHide if you get double input.

2- I'm not involved in the development of Duo so I can't speak on @Black-Seraph's behalf, but I know that windows 11 introduced many new virtualization features which Duo is capitalizing on. I'm sure these features would have been made available on Win 10 too if that was possible. Though the Patreon version is worth it for the high refresh rate alone imo.

**Comment by AnshulJ999** [comment]
Thanks for clarifying! I do own DSX and the Dualsense emulation so I will test it out and see if it might work. Hope Sunshine implements this soon so it can be passed onto Duo.

**Comment by Black-Seraph** [comment]
Let me explain why we're limited to DS4 and XBOX gamepads.

This limitations comes with a bit of history attached to it.

Input devices on Windows require a kernel driver to function, unlike, let's say monitor drivers, which can function perfectly fine off of a userspace driver.

Kernel drivers are monetized by Microsoft through their double-signing initiative though, which basically translates to pay us a lot of money to get a good boy stamp on your drivers, allowing them to load in Windows.

If your driver isn't double-signed in such a way you simply can't load and use it.

The author behind ViGEmBus forked over the required money to get his gamepad driver Microsoft-certified, which is why every software out there uses it, but ViGEmBus is abandonware, aka. no longer in active development, which is why its frozen in time in a stage where it essentially only supports DS4 or XBOX gamepad emulation.

I am trying to get myself signed up for access to a signing conclave which should, in theory, allow me to fix such things, but until that pans out we're stuck with what we've got.

**Comment by AnshulJ999** [comment]
> 
> The author behind ViGEmBus forked over the required money to get his gamepad driver Microsoft-certified, which is why every software out there uses it, but ViGEmBus is abandonware, aka. no longer in active development, which is why its frozen in time in a stage where it essentially only supports DS4 or XBOX gamepad emulation.
> 

Thanks so much for explaining all that. I've read a bit about this issue since I use DSX; and I believe the Nefarius dev now works exclusively with DSX's dev and that's why there's an exclusive Dualsense emulation driver (paid add-on) in DSX. 

Dualsense emulation in Sunshine/Moonlight/Duo would be amazing for sure. 

But, and this solution may be more specific to my case: is there any way to turn off the controller jailing? I've noticed that I can turn on DSX on my laptop and the controller will work in the game directly; but sometimes it doesn't. I suppose it might be a matter of the controller being jailed correctly/incorrectly at times. 

Basically I'd like to force the controller to not be jailed, and then, I could get the Dualsense connected to the host to be recognized on the Duo session as well. And so native Playstation games would support it. Thanks!
